primitives/src/lib.rs

```diff
@@ -67,20 +60,17 @@ pub struct ClusterBondingParams<BlockNumber> {
 #[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo, PartialEq)]
 pub enum NodePubKey {
 	StoragePubKey(StorageNodePubKey),
-	CDNPubKey(CDNNodePubKey),
```

Comment:
I am not in favor of keeping the enum even if the reason was to avoid a runtime migration.

This is probably going to be a tech debt that we have to do later so i'd rather do it in this PR instead.

Reply:
In case we have only 1 node type, I would also remove the enum. But we need to launch production most likely next week.
Removing this enum will cause a lot of changes in the integration parts, i.e.:

- JS-SDK and its e2e tests
- S&T e2e tests
- GO-SDK and its e2e tests
- DDC Storage node and its e2e test
- Blockchain unit tests in all DDC pallets
- Blockchain tests in all DDC pallets

That's why the decision was not to remove this enum at this moment, as all integration parts will not have enough time to adapt to this change.

Also, we may have other node types in DDC (AI, DAC) that can fit into this enum. The implementation of new types of nodes is at the design stage still.

Considering all the above facts, I think for now we have to leave it as the enum and migrate (in case needed) after the production launch.
Integration components are not ready for a such fundamental format change at the moment